src/fix/main.rs

```diff
@@ -0,0 +1,1059 @@
+        cli_options.rom_version = Some(rom_version)
+    }
+
+    if cli.overwrite {
+        cli_options.overwrite_rom = true;
+    }
+
+    if let Some(pad_value) = cli.pad_value {
+        cli_options.pad_value = Some(pad_value)
+    }
+
+    if let Some(ram_size) = cli.ram_size {
+        cli_options.ram_size = Some(ram_size)
+    }
+
+    if cli.sgb_compatible {
+        cli_options.sgb = true;
+    }
+
+    if let Some(title) = cli.title {
+        cli_options.title = Some(title.clone());
+        let mut len = title.len() as u8;
+        let max_len = max_title_len(&cli_options.game_id, &cli_options.model);
+        if len > max_len {
+            len = max_len;
+            println!("warning: Truncating title \"{}\" to {} chars", title, max_len);
+            cli_options.title_len = len;
+        }
+    }
+
+    if cli.color_only || cli.color_compatible {
+        cli_options.model = if cli.color_compatible { Model::Both } else { Model::Cgb };
+        if cli_options.title_len > 15 {
+            if let Some(mut title) = cli_options.title {
+                title = title[0..15].to_string();
+                eprintln!("warning: Truncating title \"{}\" to 15 chars", title);
+                cli_options.title = Some(title);
+            }
+        }
+    }
+
+    if cli.version {
+        println!("rgbfix version {}", cli.version);
+        process::exit(1);
+    }
+
+    if cli.validate {
+        cli_options.fix_spec = vec![FixSpec::FixLogo, FixSpec::FixHeaderSum, FixSpec::FixGlobalSum];
+    }
+
+    if !cli.files.is_empty() {
+        process_filename(&cli.files[0], cli_options); // TOCHECK [0] is dubious, are multiple files getting fixed in one CLI call possible?
+    }
+
+}
+
+fn report(fmt: &str, args: &[&str]) -> io::Result<()> {
+    let mut stderr = io::stderr();
+    write!(&mut stderr, "{}", fmt)?;
+    for arg in args {
+        write!(&mut stderr, " {}", arg)?;
+    }
+    writeln!(&mut stderr)?;
+    Ok(())
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum MbcType {
+    Rom,
+    RomRam,
+    RomRamBattery,
+    Mbc1,
+    Mbc1Ram,
+    Mbc1RamBattery,
+    Mbc2,
+    Mbc2Battery,
+    Mmm01,
+    Mmm01Ram,
+    Mmm01RamBattery,
+    Mbc3,
+    Mbc3TimerBattery,
+    Mbc3TimerRamBattery,
+    Mbc3Ram,
+    Mbc3RamBattery,
+    Mbc5,
+    Mbc5Ram,
+    Mbc5RamBattery,
+    Mbc5Rumble,
+    Mbc5RumbleRam,
+    Mbc5RumbleRamBattery,
+    Mbc6,
+    Mbc7SensorRumbleRamBattery,
+    PocketCamera,
+    BandaiTama5,
+    Huc3,
+    Huc1RamBattery,
+    Tpp1,
+    Tpp1Rumble,
+    Tpp1MultiRumble,
+    Tpp1MultiRumbleRumble,
+    Tpp1Timer,
+    Tpp1TimerRumble,
+    Tpp1TimerMultiRumble,
+    Tpp1TimerMultiRumbleRumble,
+    Tpp1Battery,
+    Tpp1BatteryRumble,
+    Tpp1BatteryMultiRumble,
+    Tpp1BatteryMultiRumbleRumble,
+    Tpp1BatteryTimer,
+    Tpp1BatteryTimerRumble,
+    Tpp1BatteryTimerMultiRumble,
+    Tpp1BatteryTimerMultiRumbleRumble,
+    MbcNone,
+    MbcBad,
+    MbcWrongFeatures,
+    MbcBadRange,
```

Comment:
These should probably be represented by an Option/Result, rather than being in the same enum as all the good/correct values.